### ISSUE TITLE
Seq.Concat gives incorrect results

### ISSUE THREAD

**Issue by StanJav** [louthy/language-ext#808]
The second `Seq.Concat` for lazy sequences seems to break something.

```
[Fact]
public void SeqConcatTest()
{
    var seq1 = (from x in new[] { 1 }
                select x).ToSeq();

    var seq2 = (from x in new[] { 2 }
                select x).ToSeq();


    var seq3 = (from x in new[] { 3 }
                select x).ToSeq();

    Assert.Equal(
        Seq(1, 2, 3),
        seq1
            .Concat(seq2)
            .Concat(seq3));
}
```

And result: 
```
  Message: 
    Assert.Equal() Failure
    Expected: Seq<Int32> [1, 2, 3]
    Actual:   Seq<Int32> [1, 2, 1, 2]
```

Let me know if you want me to dig deeper. 
Tested on 3.5.16-beta

**Comment by louthy** [comment]
Fixed in `v3.5.17-beta`

**Comment by louthy** [comment]
btw, if you want to concatenated multiple sequences, it's more elegant to just `+` them:

```c#
   var fseq= seq1 + seq2 + seq3;
```
